src/config.rs

```diff
@@ -63,17 +77,49 @@ pub struct Config {
     pub config_file_path: Option<PathBuf>,
 
     pub rule_configurations: Vec<RuleConfiguration>,
+
+    #[allow(clippy::type_complexity)]
+    #[builder(setter(skip))]
+    rules_by_plugin_prefixed_name: OnceLock<HashMap<String, (Arc<dyn Rule>, Option<PluginIndex>)>>,
 }
 
 impl Config {
-    fn get_active_rules(&self) -> Vec<Arc<dyn Rule>> {
+    fn get_rules_by_plugin_prefixed_name(
+        &self,
+    ) -> &HashMap<String, (Arc<dyn Rule>, Option<PluginIndex>)> {
```

Comment:
This method is assuming an invariant that the `.all_plugins` and `.all_standalone_rules` fields won't be mutated after this method is called

I think that's ok because those fields aren't `pub` and they're effectively being treated as immutable once the `Config` has been created

And so the usage of `PluginIndex` (just a `usize` index into `.all_plugins`) also assumes that invariant (since otherwise a `PluginIndex` that's getting "passed around" to keep track of which plugin a rule was associated with could become outdated)